app_backend/server_test.py

```diff
@@ -32,3 +32,35 @@ def __init__(self, uuid: str, fuuid: str, expected_status_code: int, expects_err
         response = client.post(query)
         assert response.status_code == test_case.expected_status_code
         assert ("err" in response.json()) == test_case.expects_err
+
+
+def test_getuser():
+    response = client.get("/login/")
+    assert response.status_code == 200
+    assert response.json() == {"msg": "This is the login page!"}
+
+def test_registeruser():
+    class Params:
+        def __init__(self, username: str, email: str, password: str, expects_err: bool, expects_err_msg: dict):
+            self.username = username
+            self.email = email
+            self.password = password
+            self.expects_err = expects_err
+            self.expects_err_msg = expects_err_msg
+        def to_json(self):
+            return {"username": self.username, "email": self.email, "password": self.password}
+    test_cases = [
+        Params('test000', 'test0@emory.edu', 'Test0000', status.HTTP_400_BAD_REQUEST, {"err" : "Invalid username. It must be 8 to 20 characters with no special characters."}),
+        Params('test0000', 'test0@gmail.com', 'Test0000', status.HTTP_400_BAD_REQUEST, {"err" : "Invalid email address. It must be an emory.edu email."}),
+        Params('test0000', 'test0@emory.edu', 'test0000', status.HTTP_400_BAD_REQUEST, {"err" : "Invalid password. It must be 8 to 20 characters and have at least one upper case letter, one lower case letter, and one digit."}),
+        Params('test0000', 'test@emory.edu', 'Test0000', status.HTTP_201_CREATED, {"msg": "Successfully registered new user."})
```

Comment:
I don't really care for it much at the moment, but we should add more diverse combinations here to verify the regex you built works

Reply:
i added a few more tests, hopefully could do more as I think of more test cases while finishing up the /login/ route